beacon-chain/sync/backfill/backfill.md

```diff
@@ -0,0 +1,853 @@
+- Block history queries via RPC
+- Historical state regeneration
+- Archive node capabilities
+- Data availability requirements
+- Compliance with `MIN_EPOCHS_FOR_BLOCK_REQUESTS` spec requirement
+
+### Spec Requirement
+The Ethereum specification requires nodes to serve blocks going back `MIN_EPOCHS_FOR_BLOCK_REQUESTS` epochs from the current slot. Backfill ensures this requirement is met.
+
+## Architecture
+
+The backfill system consists of several key components working together:
+
+```
+┌─────────────────────────────────────────────────────────────┐
+│                    Backfill Service                         │
+│  ┌─────────────┐  ┌──────────────┐  ┌──────────────┐        │
+│  │   Store     │  │   Verifier   │  │  Blob Store  │        │
+│  │  (Status)   │  │ (Signatures) │  │   (Deneb+)   │        │
+│  └─────────────┘  └──────────────┘  └──────────────┘        │
+│                                                             │
+│  ┌──────────────────────────────────────────────────────┐   │
+│  │           Batch Sequencer                            │   │
+│  │  - Creates batches in reverse chronological order    │   │
+│  │  - Manages batch state transitions                   │   │
+│  │  - Determines importable batches                     │   │
+│  └──────────────────────────────────────────────────────┘   │
+│                                                             │
+│  ┌──────────────────────────────────────────────────────┐   │
+│  │           Worker Pool                                │   │
+│  │  ┌──────────┐  ┌──────────┐  ┌──────────┐            │   │
+│  │  │ Worker 1 │  │ Worker 2 │  │ Worker N │            │   │
+│  │  └──────────┘  └──────────┘  └──────────┘            │   │
+│  │       ↕              ↕              ↕                │   │
+│  │  ┌──────────────────────────────────────┐            │   │
+│  │  │       Batch Router                   │            │   │
+│  │  │  - Assigns peers to batches          │            │   │
+│  │  │  - Manages worker busy state         │            │   │
+│  │  └──────────────────────────────────────┘            │   │
+│  └──────────────────────────────────────────────────────┘   │
+│                          ↕                                  │
+│  ┌──────────────────────────────────────────────────────┐   │
+│  │              P2P Network                             │   │
+│  │  - BeaconBlocksByRange requests                      │   │
+│  │  - BlobSidecarsByRange requests                      │   │
+│  └──────────────────────────────────────────────────────┘   │
+└─────────────────────────────────────────────────────────────┘
+                          ↓
+           ┌──────────────────────────────┐
+           │    Database (BeaconDB)       │
+           │  - Blocks storage            │
+           │  - Backfill status tracking  │
+           │  - Finalized index updates   │
+           └──────────────────────────────┘
+```
```

Comment:
nitpick: We also write blob | data column sidecars to their respective filesystem storage.